src/addressResolvers/index.ts

```diff
@@ -22,7 +39,11 @@ export async function lookupAddresses(addresses: Address[]) {
 
   return withoutEmptyValues(
     await cache(normalizedAddresses, async (addresses: Address[]) => {
-      const results = await Promise.all(RESOLVERS.map(r => r.lookupAddresses(addresses)));
+      const results = await Promise.all(
+        RESOLVERS.map(r =>
+          instrumentCall('lookup_addresses', r.NAME, () => r.lookupAddresses(addresses))
+        )
+      );
```

Comment:
Too confusing for noob dev like me :P to keep this combined with actual logic, is there a way to seperate instrumentCall and keep it seperate

Reply:
Will see, and ping you back when the PR is finished, I am refactoring